[JustWeiHao] iP

JustWeiHao opened the pull request:
# Raphael
> The smartest bot assistant you could possibly have
## What to expect
- Command Line UI
- ~GUI~
- Smart
## How to use
1. Get all the source code
2. Compile
3. Run

The entry point is in `Raphael.java`

## Features
- [X] Task manager
- [ ] Reminders
- [ ] [Smart](https://dictionary.cambridge.org/dictionary/english/smart) Chat Bot 👍 

Some cool stuff about the bot
- It can write the task list into a file
``` Java
BufferedWriter bw = Files.newBufferedWriter(this.filePath);
bw.write(content);
bw.close();
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

JustWeiHao committed:
Modify Duke.java

JustWeiHao committed:
Add Ui.java

JustWeiHao committed:
Modify Ui.java

JustWeiHao committed:
Modify Duke.java

JustWeiHao committed:
Modify Duke.java

JustWeiHao committed:
Modify Ui.java

JustWeiHao committed:
Add Storage.java

JustWeiHao committed:
Add Parser.java

JustWeiHao committed:
Modify Storage.java

JustWeiHao committed:
Modify Duke.java

JustWeiHao committed:
Modify Ui.java

JustWeiHao committed:
Add Task.java

JustWeiHao committed:
Add TaskList.java

JustWeiHao committed:
Add Todo.java

JustWeiHao committed:
Add Event.java

JustWeiHao committed:
Deadline.java

JustWeiHao committed:
Modify Parser.java

JustWeiHao committed:
Modify Storage.java

JustWeiHao committed:
Modify Task.java

JustWeiHao committed:
Modify TaskList.java

JustWeiHao committed:
Modify Ui.java

JustWeiHao committed:
Modify EXPECTED.TXT

JustWeiHao committed:
Modify input.txt

JustWeiHao committed:
Modify Duke.java

JustWeiHao committed:
Modify Ui.java

JustWeiHao committed:
Modify TaskList.java

JustWeiHao committed:
Modify Ui.java to respond to unknown command

JustWeiHao committed:
Modify Storage.java to support deleting item

JustWeiHao committed:
Modify TaskList.java to support deleting task

JustWeiHao committed:
Modify Ui.java to support deleting task

JustWeiHao committed:
Modify Duke.java to support reading task list from file

JustWeiHao committed:
Add FileFormattable.java to support conversion into file store format

JustWeiHao committed:
Add FileProcess.java to open, read, and write a file

JustWeiHao committed:
Modify the constructor of the classes to support creating objects read from a file

JustWeiHao committed:
Modify Parser.java to remove trailing spaces of values returned

JustWeiHao committed:
Modify Storage.java to support object that is convertable into file store format

JustWeiHao committed:
Modify TaskList.java to support loading and storing into file

JustWeiHao committed:
Merge branch 'branch-level-7'

JustWeiHao committed:
Modify Ui.java to implement new command to print deadlines on specific date

JustWeiHao committed:
Modify Storage.java to support printing deadline is due by specific due

JustWeiHao committed:
Modify Parser.java to check for date format using simpler code

JustWeiHao committed:
Modify files to change representation of date from String to LocalDate

JustWeiHao committed:
Merge branch 'branch-level-8'

JustWeiHao committed:
Refactor code v1

JustWeiHao committed:
Refactor code v2

JustWeiHao committed:
Refactor code v3

JustWeiHao committed:
Debug to write file after check/uncheck task

JustWeiHao committed:
Merge branch 'MoreOOP'

JustWeiHao committed:
Reorganize classes into packages

JustWeiHao committed:
Merge branch 'Packages'

JustWeiHao committed:
Merge branch 'add-gradle-support'

JustWeiHao committed:
Change package name

JustWeiHao committed:
Change DukeException to RaphaelException

JustWeiHao committed:
Add AddCommandTest.java

JustWeiHao committed:
Add DeleteCommandTest.java

JustWeiHao committed:
Add EditCommandTest.java

JustWeiHao committed:
Add ExitCommandTest.java

JustWeiHao committed:
Modify build.gradle

JustWeiHao committed:
Merge branch 'branch-A-JUnit'

JustWeiHao committed:
Add JavaDoc v1

JustWeiHao committed:
Add JavaDoc v2

JustWeiHao committed:
Merge branch 'branch-A-JavaDoc'

JustWeiHao committed:
Refactor code using enum

JustWeiHao committed:
Solve merge conflict

JustWeiHao committed:
Add Find function

JustWeiHao committed:
Resolve merge conflict

JustWeiHao committed:
Add TaskTest.java

JustWeiHao committed:
Modify AddCommandTest.java

belligerentbeagle commented on src/main/java/raphael/Raphael.java:
```diff
@@ -0,0 +1,61 @@
+            + " | | \\ \\| (_| || |_) || | | || (_| ||  __/| |\n"
+            + " |_|  \\_\\\\__,_|| .__/ |_| |_| \\__,_| \\___||_|\n"
+            + "               | |\n"
+            + "               |_|\n";
```
Cool text art!

belligerentbeagle commented on src/main/java/raphael/Raphael.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    private void run() {
+        this.ui.showWelcome();
+        boolean isExit = false;
```
great use of appropriate variable names

belligerentbeagle commented on src/main/java/raphael/Raphael.java:
```diff
@@ -0,0 +1,61 @@
+                isExit = c.isExit();
+            } catch (raphael.exception.RaphaelException e) {
+                ui.showError(e.getMessage());
+            } finally {
```
great use of try catch finally logic!

belligerentbeagle commented on src/main/java/raphael/command/Command.java:
```diff
@@ -0,0 +1,46 @@
+        public String getFormat() {
+            return this.format;
+        }
+    }
```
perhaps an additional line here?

belligerentbeagle commented on src/main/java/raphael/exception/RaphaelException.java:
```diff
@@ -0,0 +1,32 @@
+     * @return the error message.
+     */
+    public static String invalidFormat(Command.TYPE commandType) {
+        return String.format("Please follow the format!:\n\t%s", commandType.getFormat());
```
good regex use!

belligerentbeagle commented on src/main/java/raphael/parser/Parser.java:
```diff
@@ -0,0 +1,103 @@
+        }
+        final String[] inputArr = input.split(" ", 2);
+        switch(inputArr[0]) {
+            case "bye":
```
Not sure if the case in switch case should be further indented or on the same line according to 2103's coding standard?

belligerentbeagle commented on src/main/java/raphael/parser/Parser.java:
```diff
@@ -0,0 +1,103 @@
+                throw new RaphaelException("I'm sorry that I can't recognize the command!");
+        }
+    }
+    private static Command processTaskWithIndex(String[] inputArr, Command.TYPE commandType) throws RaphaelException {
```
good method naming!

belligerentbeagle commented on src/main/java/raphael/task/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+    private final LocalDateTime deadline;
+    private final DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private final DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("MMM dd yyyy HHmm");
+    public Deadline(String description, String deadline) throws RaphaelException {
```
perhaps leave a line spacing between each method?

belligerentbeagle commented on src/main/java/raphael/task/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+        for (String task : tasksArr) {
+            final String[] taskArr = task.split(" \\|&\\| ");
+            switch (taskArr[0]) {
+                case "T":
```
Same comment on the switch case indentation as mentioned before

belligerentbeagle commented on src/main/java/raphael/task/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+        }
+        return fileFormat;
+    }
+}
```
overall Good coding standard!

belligerentbeagle commented on src/main/java/raphael/ui/Ui.java:
good!

belligerentbeagle reviewed (commented):
Overall good coding standard, LGTM. Just require a couple nit fixes.

JustWeiHao committed:
Fix parser checkstyle

JustWeiHao committed:
Fix Todo checkstyle

JustWeiHao committed:
Fix Deadline.java checkstyle

JustWeiHao committed:
Fix Ui.java checkstyle

JustWeiHao committed:
Fix Command.java checkstyle

JustWeiHao committed:
Fix Event.java checkstyle

JustWeiHao committed:
Fix Task.java checkstyle

JustWeiHao committed:
Fix RaphaelException.java checkstyle

JustWeiHao committed:
Fix Raphael.java checkstyle

JustWeiHao committed:
Fix TaskList.java checkstyle

JustWeiHao committed:
Fix CheckCommand.java checkstyle

JustWeiHao committed:
Fix DeleteCommand.java checkstyle

JustWeiHao committed:
Fix EditCommand.java checkstyle

JustWeiHao committed:
Fix ExitCommand.java checkstyle

JustWeiHao committed:
Fix Storage.java checkstyle

JustWeiHao committed:
Fix AddCommand.java checkstyle

JustWeiHao committed:
Fix FindCommand.java checkstyle

JustWeiHao committed:
Fix ListCommand.java checkstyle

JustWeiHao committed:
Fix FileFormattable.java checkstyle

JustWeiHao committed:
Fix all checkstyle

JustWeiHao committed:
Fix checkstyle

JustWeiHao committed:
Merge branch 'branch-A-Checkstyle'

JerryWang0000 commented on src/main/java/raphael/Raphael.java:
```diff
@@ -0,0 +1,71 @@
+ * The main class of Raphael.
+ */
+public class Raphael {
+    public static final String LOGO = "\n"
```
Very Artistic and I like it too.

JerryWang0000 commented on src/main/java/raphael/Raphael.java:
```diff
@@ -0,0 +1,71 @@
+            + " |_|  \\_\\\\__,_|| .__/ |_| |_| \\__,_| \\___||_|\n"
+            + "               | |\n"
+            + "               |_|\n";
+    public static final String BOT_NAME = "raphael";
```
I like the naming as it conforms to the coding standard, as it is already confirmed here.

JerryWang0000 commented on src/main/java/raphael/command/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws raphael.exception.RaphaelException {
+        tasks.addTask(this.toAdd);
+        ui.showAddOutput(tasks, this.toAdd);
+        storage.write(tasks.toFileFormat());
```
I like the command-like naming in the methods. It adds readability.

JerryWang0000 commented on src/main/java/raphael/command/Command.java:
```diff
@@ -0,0 +1,52 @@
+        LIST("list"),
+        CHECK("check [dd/MM/yyyy HHmm]"),
+        FIND("find [word]");
+        private final String format;
```
Like your use of enum to refer to all the commands.

JerryWang0000 commented on src/main/java/raphael/command/EditCommand.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public class EditCommand extends Command {
+    private final int idx;
+    private final boolean isMark;
```
I would write isMarked

JerryWang0000 commented on src/main/java/raphael/format/FileFormattable.java:
Very inspiring to package the formatting related files.

JerryWang0000 commented on src/main/java/raphael/task/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+     * @param isDone the status of the deadline task.
+     * @throws RaphaelException exception exclusive to Raphael.
+     */
+    public Deadline(String description, String deadline, boolean isDone) throws RaphaelException {
```
I also prefer this signiture as the specialized Exception would be easier to handle.

JerryWang0000 commented on src/main/java/raphael/task/Event.java:
```diff
@@ -0,0 +1,68 @@
+                this.from.format(this.outputFormat),
+                this.to.format(this.outputFormat));
+    }
+    @Override
```
A blank line here?

JerryWang0000 commented on src/main/java/raphael/task/Todo.java:
Perhaps these codes are too compact? IDE can save the work by just 1 copy paste.

JerryWang0000 commented on src/main/java/raphael/ui/Ui.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * Prints the line indicating that the following output is an error message.
+     */
+    private static void printWarningLine() {
```
I like how you add JavaDoc even to private methods.

JerryWang0000 commented on text-ui-test/EXPECTED.TXT:
I personally ignored this file but I'm not quite sure whether it should be revision controlled.

JerryWang0000 reviewed (commented):
I like your usage of packages especially as they make the project clearer.

JustWeiHao committed:
Add GUI v1

JustWeiHao committed:
Add GUI

JustWeiHao committed:
Merge branch 'branch-Level-10'

JustWeiHao committed:
Add assertions

There are some assertions statement added to ensure the expected
properties of the object.

Parts that have assertions included:
* Methods of Task that modify the field of the task.
* Methods of TaskList tht adding and deleting the task.

JustWeiHao committed:
Refactor Code

Some changes are made on the methods name.
* Change the use of check and uncheck to mark and unmark.
* The methods name are more informative.

JustWeiHao committed:
Introduce stream

Introduce the usage of stream to replace loop.
* Finding tasks that contain the given keyword.
* Coversion of task list into the correct file format.
* Sort the task list according to alphabetical order.

JustWeiHao committed:
Merge pull request #1 from JustWeiHao/branch-A-Assertions

Add assertions

JustWeiHao committed:
Merge pull request #3 from JustWeiHao/branch-A-CodeQuality

Refactor Code

JustWeiHao committed:
Merge branch 'master' into branch-A-Streams

JustWeiHao committed:
Merge pull request #2 from JustWeiHao/branch-A-Streams

Introduce stream

JustWeiHao committed:
Add file for CI

A gradle.yml is added for Continuous Integration of Raphael.

JustWeiHao committed:
Fix bug

* Insert the missing curly brace
* Update the test cases for exit command to support the new
  implementation.

JustWeiHao committed:
Fix Checkstyle

Multiple files have been change to follow the Java Coding style.

JustWeiHao committed:
Add ability to detect duplicate task

The bot will now throw exception if the exact same task is being added
twice. The Task class and its subclasses now override Object::equals to
support the checking of duplicate tasks.

JustWeiHao committed:
Merge branch 'branch-C-DetectDuplicates'

JustWeiHao committed:
Modify test cases

All the test cases will modify a test file instead of the actual tasks
file.

JustWeiHao committed:
Merge branch 'branch-A-MoreTesting'

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Enable resizable GUI

JustWeiHao committed:
Add hello command

JustWeiHao committed:
Merge branch 'branch-A-UserGuide' of https://github.com/JustWeiHao/ip into branch-A-UserGuide

JustWeiHao committed:
Update User Guide

User Guide is updated with full sets of command.

JustWeiHao committed:
Fix JavaDocs

JustWeiHao committed:
Update user guide

JustWeiHao committed:
Merge branch 'branch-A-UserGuide'

JustWeiHao committed:
Add RaphaelTest.java

JustWeiHao committed:
Modify build.gradle

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Update README.md

JustWeiHao committed:
Fix linking to header

JustWeiHao committed:
Merge branch 'branch-A-UserGuide'

JustWeiHao committed:
Update user guide

JustWeiHao committed:
Merge branch 'branch-A-UserGuide'

JustWeiHao committed:
Improve gui

JustWeiHao committed:
Update fxml for better gui

JustWeiHao committed:
Merge branch 'branch-A-BetterGUI'

JustWeiHao committed:
Fix checkstyle

JustWeiHao committed:
Add help command

JustWeiHao committed:
Update gui pictures
